src/triple.rs

```diff
@@ -70,6 +69,70 @@ pub enum CallingConvention {
     WindowsFastcall,
 }
 
+/// The C data model used on a target.
+///
+/// See also https://en.cppreference.com/w/c/language/arithmetic_types
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[non_exhaustive]
+pub enum CDataModel {
+    /// The data model used most commonly on Win16
+    LP32,
+    /// The data model used most commonly on Win32 and 32-bit Unix systems
+    ILP32,
+    /// The data model used most commonly on Win64
+    LLP64,
+    /// The data model used most commonly on 64-bit Unix systems
+    LP64,
+    /// A rare data model used on early 64-bit Unix systems
+    ILP64,
+}
+
+impl CDataModel {
+    /// The width of a pointer (in the default address space).
+    pub fn pointer_width(&self) -> Size {
+        match self {
+            CDataModel::LP32 | CDataModel::ILP32 => Size::U32,
+            CDataModel::LLP64 | CDataModel::LP64 | CDataModel::ILP64 => Size::U64,
```

Comment:
`non_exhaustive` only impacts downstream users, not the library it's defined in. Personally this is expected behavior for me, this ensures that if we add a new `CDataModel` variant we'll have to add it to `pointer_width()`, etc. before publishing. This is the same reason I used a match in `long_long_size` instead of a plain `CDataModel::U64`.

Reply:
Ah, that make sense!